tests/poll.rs

```diff
@@ -112,8 +111,7 @@ fn test_registry_behind_arc() {
         barrier3.wait();
     });
 
-    poll.poll(&mut events, Some(Duration::from_millis(100)))
-        .unwrap();
+    poll.poll(&mut events, None).unwrap();
```

Comment:
Can keep the timeout, if needed you can increase the timeout (max 1 second or so), I don't tests to get stuck.